core/unix/rseq_linux.c

```diff
@@ -78,6 +78,18 @@ DECLARE_CXTSWPROT_VAR(static mutex_t rseq_trigger_lock,
                       INIT_LOCK_FREE(rseq_trigger_lock));
 static volatile bool rseq_enabled;
 
+/* The struct rseq registered by glibc is present in the struct pthread.
+ * As of glibc 2.35, it is present at the following offset from the app
+ * lib seg base. We check these offsets first and then fall back to a
+ * wider search. The linux.rseq test helps detect changes in these
+ * offsets in future glibc versions.
+ */
+#ifdef X86
+#    define GLIBC_RSEQ_OFFSET 2464
```

Comment:
Is this different for 32-bit vs 64-bit?

Reply:
Ah. I added the offset for x86-32 and also tested it. But I don't have access to an AArch32 machine to verify for 32-bit arm. Added a comment for now.

Reply:
Our rseq tests are currently for 64-bit only. But to get just the offset, maybe we can run a simple program under Qemu. But, the cross toolchain that's available to me is at 2.33 for AArch32, but I need 2.35+.

Since we have the fallback too, the XXX comment should do for now.